docs/workflows/best-practices-osh.md

```diff
@@ -0,0 +1,398 @@
+
+
+### 7. Handling detected lack of freshness [ROUGH DRAFT]
+- The teams are responsible for keeping libraries reasonably up to date: 
+  - small library updates can be updated as part of regular maintenance; 
+  - larger updates (e.g. major versions or frameworks) should be planned explicitly as part of a development sprint.
+
+### 8. _Handling detected lack of activity_ (TBD)
+
+## Handling your libraries
+
+
+### 9. Updating a library [ROUGH DRAFT]
+There can be several reasons to consider updating a library.
+
+Checklist before updating:
+- Does it have open issues?​
+- Is the license (still) compatible?​
+  - → [Handling detected license issues](#handling-detected-license-issues)
+- Does it contain optimizations for performance, maintainability, functionality?​
+  - ? not sure why this is exactly asked?
+- Is the new version compatible?  ​
+- How mature is the version?​
+- Is it stable enough?​
+
+Always Manage libraries with a package manager (don’t mix third party code with first party code)​
+
+- compatibility break
+  - Compatibility breaks can be caused by the framework/library no longer supporting the technologies used in the system.​
+  - Analyze the impact and consider the following things:​
+  - What is the impact of updating the system to make it compatible?​
+  - Are there alternatives of the library?​
+  - Are the technologies used still relevant for the future of the system?​
+
+- when is postponing updates a good idea?
+  - The longer you postpone updating, the bigger the eventual pain. As your system grows and evolves, the costs and risks of upgrading an old library increase. Such an accumulation of maintenance debt may lead to a much larger effort than in the case of smaller incremental updates.​
+  - For core functionalities, it is a good idea to stay on a stable version of the library. ​Being an early adopter of a new version comes with some risks. Bugs are common in immature versions of libraries as they are not thoroughly tested. Although you have the opportunity to contribute, by testing and potentially providing solutions to bugs you encounter in open source libraries, it might be a better practice to wait until the library becomes more mature.​​
+  - Don’t go for the "If it ain’t broke, don’t fix it" strategy​
+    - This strategy implies that you don’t update unless you have to. You stay with the current version of the third-party library until you notice something wrong in your application, no matter how often the vendor publishes an update. ​
+    - Whilst easier in the short term, with this strategy you will end up with a system which depends on outdated and unmaintained libraries, where you can't use some other libraries since they require a newer version of that library which you can't upgrade and at some point. You may lose the ability to fix some issues at all.​
+
+
+### 10. Adopting a new library [ROUGH DRAFT]
```

Comment:
I'd use the word 'use' instead of 'adopt'. We use 'adopting' elsewhere to indicate taking responsibility for maintaining the code, which is not what we try to say here, right?